### PR TITLE
Fix the connexion to the side of the repeater

### LINKED ISSUE #4639: Lamp gets turned off when a repeater is placed diagonally
Client version: 12.2.2
Server OS: Linux
Cuberite Commit id: 0d722701861ad9acfa7d36c5156fd184e3ba6bee

### Expected behavior
The lamp doesn't care about a repeater diagonally.

### Actual behavior
See screenshots:

### Steps to reproduce the behavior
Place everything down as shown on the screenshots.

### Server log
```
-
```

### PR DESCRIPTION
**Pull request by mBornand**
The changes check if the repeater is connected to the side or not.
Fixes #4639

### CODE REVIEW

File: src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h [hunk 1]
```diff
@@ -14,12 +14,34 @@ class cRedstoneWireHandler:
...
 public:
 
-	inline static bool IsDirectlyConnectingMechanism(BLOCKTYPE a_Block)
+	inline static bool IsDirectlyConnectingMechanism(cWorld & a_World, BLOCKTYPE a_Block, Vector3i a_QueryPosition, Vector3i a_Offset)
 	{
 		switch (a_Block)
 		{
 			case E_BLOCK_REDSTONE_REPEATER_ON:
 			case E_BLOCK_REDSTONE_REPEATER_OFF:
+				if ((a_World.GetBlockMeta(a_QueryPosition) & 0x1) == 1)
```
Comment (bearbin) [inline src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h:23]: This line needs its magic numbers replacing with either named constants or, better, a function that already exists with semantic meaning. Some comments for what the two cases mean are also necessary.

File: src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h [hunk 2]
```diff
@@ -14,12 +14,34 @@ class cRedstoneWireHandler:
 
 public:
 
-	inline static bool IsDirectlyConnectingMechanism(BLOCKTYPE a_Block)
+	inline static bool IsDirectlyConnectingMechanism(cWorld & a_World, BLOCKTYPE a_Block, Vector3i a_QueryPosition, Vector3i a_Offset)
```
Comment (bearbin) [inline src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h:17]: This function should just take `a_Block`, `a_BlockMeta` and `a_Offset`, there's no reason to have any side effects depending on a world object.

**Comment by bearbin** [comment]
Please also cherry-pick your commit adding yourself as a contributor into this branch, so they can be merged together.

File: src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h [hunk 3]
```diff
@@ -14,12 +14,39 @@ class cRedstoneWireHandler:
...
+				{
+					/** Check if Block is not aligned in in x direction */
+					if (a_Offset.x == 0)
+					{
+						return false;
+					}
+					else
+					{
+						return true;
+					}
```
Comment (peterbell10) [inline src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h:35]:
```suggestion
					return (a_Offset.x != 0);
```

File: src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h [hunk 4]
```diff
@@ -14,12 +14,39 @@ class cRedstoneWireHandler:
...
+				{
+					/** Check if block is not aligned in z direction */
+					if (a_Offset.z == 0)
+					{
+						return false;
+					}
+					else
+					{
+						return true;
+					}
```
Comment (peterbell10) [inline src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h:48]:
```suggestion
					return (a_Offset.z != 0);
```

**Comment by madmaxoft** [comment]
Guys, stylechecking! We don't want doxy-comments inside code, and we want braces for `case` branches.

**Comment by tigerw** [comment]
sorry sorry, don't worry I'll fix it later

**Pull request merged**